docs/RFCS/typing.md

```diff
@@ -357,16 +357,19 @@ a preference to the type on the right".
 
 This is different from casts, see below.
 
-The need for this type of extension was also part of both Rick and Morty.
+The need for this type of extension is also implicitly
+present/expressed in the alternate proposals Rick and Morty.
 
-### First pass: placeholder annotation assertion
+### First pass: placeholder annotations
 
-For the first pass, we walk through the expression tree and assign the placeholders
-that are direct arguments of explicit type annotations to those types.
+For the first pass, we walk through the expression tree and:
 
-(= for each placeholder, if there is a type annotation around it,
-assign this annotation as the placeholder's type before the rest of type checking
-starts; fail if another earlier annotation provides a conflicting type.)
+- for each placeholder, check if every occurrence of that placeholder
+  is an immediate argument of a cast expression towards the same type.
+  If so, then assign the type demanded by the cast(s) to the placeholder.
```

Comment:
What happens if some of the placeholders are arguments of cast expressions and some are arguments of explicit type annotations? This should probably be specified, as it might force the two phases to split.